### PR TITLE
[B2BP-316] Prevent ".html" suffix being added to URLs ending in ".woff2"

### PR DESCRIPTION
**Pull request by Meolocious**
#### List of Changes
- Fixed the file `viewer-request-handler.ts` to handle requests, preventing the addition of ".html" to ".woff2" files.
- Updated the `viewer-request-handler.test.ts` test file to align with the recent changes.

#### Motivation and Context
This change is necessary to address a specific issue where the `file viewer-request-handler.ts` was incorrectly handling requests, leading to the unintended addition of ".html" to ".woff2" files. The fix ensures that this behavior is corrected, providing a more reliable file handling mechanism.

#### How Has This Been Tested?
I have thoroughly tested these changes in a controlled environment. The testing involved checking the behavior of the `viewer-request-handler.ts` when handling requests related to ".woff2" files. Additionally, the `viewer-request-handler.test.ts` file was updated to reflect the recent changes and to ensure the accuracy of the tests.

Testing Environment:
- Operating System: Windows 10
- Node.js version: 20.10.0

#### Types of changes
- [ ] Chore (nothing changes from a user perspective)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Doesn't need any documentataion changes.

### CODE REVIEW

File: apps/cloudfront-functions/src/viewer-request-handler.ts [hunk 1]
```diff
@@ -5,16 +5,20 @@
 /* eslint-disable functional/no-expression-statements */
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const handler = (
+export const handler = (
```
Comment (kin0992) [inline apps/cloudfront-functions/src/viewer-request-handler.ts:8]: You cannot use the `export` keyword here. Doing this, you are kind of reverting [this PR that fixed the issue](https://github.com/pagopa/b2b-portals/pull/125).

If you use the `export` there, you will get a runtime error on the function

File: apps/cloudfront-functions/package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "compile": "npm run clean && tsc",
     "lint": "eslint src",
     "pretest": "npm run compile",
-    "test": "vitest"
+    "test": "npm run compile && vitest"
```
Comment (kin0992) [inline apps/cloudfront-functions/package.json:10]: This is not necessary: there is the `pretest` script that does what you want.

Reference: https://docs.npmjs.com/cli/v9/using-npm/scripts#pre--post-scripts

**Review by kin0992 — CHANGES_REQUESTED** [review]

File: apps/cloudfront-functions/src/viewer-request-handler.ts [hunk 3]
```diff
@@ -13,8 +13,12 @@ const handler = (
     const { request } = event;
     const uri = request.uri;
 
-    // Add the .html extension if missing
-    if (!uri.endsWith('/') && !/\.[a-zA-Z]+$/.test(uri)) {
+    // Add the .html extension if missing and the uri does not end with .woff2
+    if (
+      !uri.endsWith('/') &&
+      !uri.endsWith('.woff2') &&
+      !/\.[a-zA-Z]+$/.test(uri)
```
Comment (simosalsi) [inline apps/cloudfront-functions/src/viewer-request-handler.ts:20]:
```suggestion
      !/\.[0-9a-zA-Z]+$/.test(uri)
```

I would just make this more generic and prevent the program from adding '.html' to any url ending in any extension which contains letters or numbers (hence including .woff2)

**Review by simosalsi — CHANGES_REQUESTED** [review]

File: apps/cloudfront-functions/src/__tests__/viewer-request-handler.test.ts [hunk 4]
```diff
@@ -35,9 +35,10 @@ describe('handler', () => {
     expect(handler(makeEvent('/page')).uri).toBe('/page.html');
     expect(handler(makeEvent('/page/')).uri).toBe('/page/');
     expect(handler(makeEvent('/image.jpg')).uri).toBe('/image.jpg');
+    expect(handler(makeEvent('/font.woff2')).uri).toBe('/font.woff2'); // new test case
     const example0 = '/i-s/g/mo/v1.0/i-p/p-i/v-d';
     expect(handler(makeEvent(example0)).uri).toBe(`${example0}.html`);
     const example1 = '/i-s/g/mo/v1.0';
-    expect(handler(makeEvent(example1)).uri).toBe(`${example1}.html`);
+    expect(handler(makeEvent(example1)).uri).toBe(example1);
```
Comment (simosalsi) [inline apps/cloudfront-functions/src/__tests__/viewer-request-handler.test.ts:42]: Is this the desired behaviour?

I think it's fine, but I just wanted to gather consensus.

File: apps/cloudfront-functions/src/viewer-request-handler.ts [hunk 5]
```diff
@@ -13,8 +13,12 @@ const handler = (
     const { request } = event;
     const uri = request.uri;
 
-    // Add the .html extension if missing
-    if (!uri.endsWith('/') && !/\.[a-zA-Z]+$/.test(uri)) {
+    // Add the .html extension if missing and the uri does not end with .woff2
+    if (
+      !uri.endsWith('/') &&
+      !uri.endsWith('.woff2') &&
```
Comment (simosalsi) [inline apps/cloudfront-functions/src/viewer-request-handler.ts:19]:
```suggestion
```
Redundant

**Review by simosalsi — CHANGES_REQUESTED** [review]

File: apps/cloudfront-functions/src/viewer-request-handler.ts [hunk 6]
```diff
@@ -13,8 +13,8 @@ const handler = (
     const { request } = event;
     const uri = request.uri;
 
-    // Add the .html extension if missing
-    if (!uri.endsWith('/') && !/\.[a-zA-Z]+$/.test(uri)) {
+    // Add the .html extension if missing and the uri does not end with .woff2
```
Comment (simosalsi) [inline apps/cloudfront-functions/src/viewer-request-handler.ts:16]:
```suggestion
    // Add the .html extension if missing
```
I don't think there's a need to specify .woff2. It shows from tests.

**Review by simosalsi — APPROVED** [review]

File: .changeset/perfect-cycles-care.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+"cloudfront-functions": patch
+---
+
+manage the ".html" extension to not be added improperly
```
Comment (simosalsi) [inline .changeset/perfect-cycles-care.md:5]:
```suggestion
prevent the ".html" extension being added to any url ending in an extension (.jpg, .html, .woff2, etc...)
```
I would go for something a little more specific as per suggestion.

**Review by datalek — APPROVED** [review]

**Pull request merged**